climbing_stairs/climbing_stairs.py

```diff
@@ -1,10 +1,20 @@
 #!/usr/bin/python
 import sys
-
+#cache holds previous values
 def climbing_stairs(n, cache=None):
-  pass 
-
+  if n <=  1:
+    return 1
+  if n == 2: 
+    return 2
+  if n == 3:
+    return 4
+  return climbing_stairs(n - 1) + climbing_stairs(n-2) + climbing_stairs(n-3)
```

Comment:
great very clean first pass. you define the base cases and the condition to terminate the recursion.